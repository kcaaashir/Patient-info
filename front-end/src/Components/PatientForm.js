import React from "react";
import * as Yup from "yup";
import { Formik, Form, Field, ErrorMessage, } from "formik";
import { FormGroup, Button } from "react-bootstrap";

const PatientForm = (props) => {
  const phoneRegExp =
    /^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/;
  const validationSchema = Yup.object().shape({
    fullname: Yup.string().required("Required"),
    email: Yup.string().email("You have entered an invalid email address."),
    phoneNo: Yup.string()
      .matches(phoneRegExp, "Invalid phone number.")
      .required("Required"),
    dateOfBirth: Yup.date().required("Required"),
    address: Yup.string().required("Required"),
    gender: Yup.string().required("Required"),
    specialAttention: Yup.boolean().required("Required"),
    file: Yup.mixed().optional(),
  });
  return (
    <div className="form-wrapper">
      <Formik {...props} validationSchema={validationSchema} >
        {(formik) => {
          return (
            <>
              <Form>
                <FormGroup className="mb-2">
                  <label>Fullname:</label>
                  <Field name="fullname" type="text" className="form-control" />
                  <ErrorMessage
                    name="fullname"
                    className="d-block invalid-feedback"
                    component="span"
                  />
                </FormGroup>
                <FormGroup className="mb-2">
                  <label>Email:</label>
                  <Field name="email" type="text" className="form-control" />
                  <ErrorMessage
                    name="email"
                    className="d-block invalid-feedback"
                    component="span"
                  />
                </FormGroup>
                <FormGroup className="mb-2">
                  <label>Phone Number:</label>
                  <Field name="phoneNo" type="number" className="form-control" />
                  <ErrorMessage
                    name="phoneNo"
                    className="d-block invalid-feedback"
                    component="span"
                  />
                </FormGroup>
                <FormGroup className="mb-2">
                  <label>Address:</label>
                  <Field name="address" type="text" className="form-control" />
                  <ErrorMessage
                    name="address"
                    className="d-block invalid-feedback"
                    component="span"
                  />
                </FormGroup>
                <FormGroup className="mb-2">
                  <label>Gender:</label>
                  <Field name="gender" as="select" className="form-control">
                    <option value="">Please select value</option>
                    <option value="MALE">Male</option>
                    <option value="FEMALE">Female</option>
                  </Field>
                  <ErrorMessage
                    name="gender"
                    className="d-block invalid-feedback"
                    component="span"
                  />
                </FormGroup>
                <FormGroup className="mb-2">
                  <label>Special Attention:</label>
                  <Field name="specialAttention" as="select" className="form-control" >
                    <option value="">Please select value</option>
                    <option value="true">Yes</option>
                    <option value="false">No</option>
                  </Field>
                  <ErrorMessage
                    name="specialAttention"
                    className="d-block invalid-feedback"
                    component="span"
                  />
                </FormGroup>
                <FormGroup className="mb-3">
                  <label>Date Of Birth:</label>
                  <Field name="dateOfBirth" type="date" className="form-control" />
                  <ErrorMessage
                    name="dateOfBirth"
                    className="d-block invalid-feedback"
                    component="span"
                  />
                </FormGroup>
                <FormGroup className="mb-3">
                  <label>Select Image</label>
                  <input
                    type="file"
                    name="file"
                    onChange={e => {
                      formik.setFieldValue('file', e.target.files[0]);
                    }}

                    className="form-control"
                  />
                  <ErrorMessage
                    name="file"
                    className="d-block invalid-feedback"
                    component="span"

                  />
                </FormGroup>
                <Button variant="primary" size="lg" block="block" type="submit">
                  {props.children}
                </Button>
              </Form>
            </>
          )
        }}
      </Formik>
    </div>
  );
};

export default PatientForm;
