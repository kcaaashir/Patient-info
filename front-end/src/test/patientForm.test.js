import React from "react";
import * as Yup from "yup";
import { Formik, Form, Field, ErrorMessage } from "formik";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import PatientForm from "../Components/PatientForm";

describe("PatientForm", () => {

  let container = null;
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("renders correctly", () => {
    act(() => {
      render(<PatientForm />, container);
    });
    expect(container.innerHTML).toMatchSnapshot();
  });

  
});
