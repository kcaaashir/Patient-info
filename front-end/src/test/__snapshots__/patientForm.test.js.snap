// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientForm renders correctly 1`] = `"<div class=\\"form-wrapper\\"><form action=\\"#\\"><div class=\\"mb-2\\"><label>Fullname:</label><input name=\\"fullname\\" type=\\"text\\" class=\\"form-control\\" value=\\"\\"></div><div class=\\"mb-2\\"><label>Email:</label><input name=\\"email\\" type=\\"text\\" class=\\"form-control\\" value=\\"\\"></div><div class=\\"mb-2\\"><label>Phone Number:</label><input name=\\"phoneNo\\" type=\\"number\\" class=\\"form-control\\" value=\\"\\"></div><div class=\\"mb-2\\"><label>Address:</label><input name=\\"address\\" type=\\"text\\" class=\\"form-control\\" value=\\"\\"></div><div class=\\"mb-2\\"><label>Gender:</label><select name=\\"gender\\" class=\\"form-control\\"><option value=\\"\\">Please select value</option><option value=\\"MALE\\">Male</option><option value=\\"FEMALE\\">Female</option></select></div><div class=\\"mb-2\\"><label>Special Attention:</label><select name=\\"specialAttention\\" class=\\"form-control\\"><option value=\\"\\">Please select value</option><option value=\\"true\\">Yes</option><option value=\\"false\\">No</option></select></div><div class=\\"mb-3\\"><label>Date Of Birth:</label><input name=\\"dateOfBirth\\" type=\\"date\\" class=\\"form-control\\" value=\\"\\"></div><div class=\\"mb-3\\"><label>Select Image</label><input type=\\"file\\" name=\\"file\\" class=\\"form-control\\"></div><button type=\\"submit\\" block=\\"block\\" class=\\"btn btn-primary btn-lg\\"></button></form></div>"`;
